cranelift-wasm/src/environ/spec.rs

```diff
@@ -378,4 +378,10 @@ pub trait ModuleEnvironment<'data> {
         offset: usize,
         data: &'data [u8],
     );
+
+    /// Declares the name of a function.
+    ///
+    /// By default this does nothing, but implementations can use this to read
+    /// the name section if desired.
+    fn declare_func_name(&self, _func_index: FuncIndex, _name: &'data str) {}
```

Comment:
By default this does nothing, since the name section is an optional custom section.